torch/csrc/autograd/record_function_ops.cpp

```diff
@@ -30,6 +31,8 @@ static void record_function_enter(
       rec.before(name);
     }
   }
+  // Record the annotation into CCA for memory snapshot tool.
```

Comment:
`rec.before` seems like an existng facility for doing callbacks when a recording starts that this should probably be using. Then the actual thing can be registered from a place that can import the cuda header, and we can avoid having to put functions in all the other allocator.s

Reply:
Great point! actually that saves use a lot of hacky static function callbacks, this facility seems much easier to read and maintain. Thanks!